### DISCUSSION TITLE
Reference R package location when loading on mox

### DISCUSSION THREAD

**Discussion by yaaminiv** [RobertsLab/resources#1187] [Q&A] [answered]
I'm running [this `methylKit` script](https://github.com/RobertsLab/project-gigas-oa-meth/blob/master/code/06-methylKit-mox.sh) on mox and I keep running into the following errors when loading all my packages:

![Screen Shot 2021-04-22 at 9 43 51 AM](https://user-images.githubusercontent.com/22335838/115752740-4036a900-a34f-11eb-92e0-d96f1a30c4b7.png)

![Screen Shot 2021-04-22 at 9 42 41 AM](https://user-images.githubusercontent.com/22335838/115752575-15e4eb80-a34f-11eb-9fa9-a1b1afaa7c31.png)

I know I installed these packages, so I think they're not being installed from their actual location? `BiocManager`, `devtools`, and `dplyr` are in the `/gscratch/srlab/rpackages/` directory:

![Screen Shot 2021-04-22 at 9 47 59 AM](https://user-images.githubusercontent.com/22335838/115753238-d4087500-a34f-11eb-81f2-6a0cd9ce2cea.png)

`methylKit` is installed in `/gscratch/home/yaaminiv/R/x86_64-pc-linux-gnu-library/3.6/`:

![Screen Shot 2021-04-22 at 9 50 32 AM](https://user-images.githubusercontent.com/22335838/115753561-2ea1d100-a350-11eb-8705-07727c122f6f.png)

## Is there a way to reference the package's location when loading the package with `require()`?

**Comment by kubu4** [comment, marked as answer]
Try setting the library path in an R environment file.

Check the contents of your `.Renviron` file:

`cat ~/.Renviron `

This will show you if there's already a set library location. If that file doesn't exist, create it and then add this to it:

`R_LIBS="/gscratch/home/yaaminiv/R/"`

I'm not entirely sure if that will be recursive or not. You may have to update it to be:

`R_LIBS="/gscratch/home/yaaminiv/R/x86_64-pc-linux-gnu-library/3.6/"`

**Reply by kubu4** [reply]
Ugh, sorry, just realized you have packages in two different locations!

If you want to load a specific package from a specific location, you can use this example:

`library(mypkg, lib.loc = "/gscratch/home/yaaminiv/R/x86_64-pc-linux-gnu-library/3.6/")`

**Reply by yaaminiv** [reply]
sweet! that argument exists for `require()` too. I made the following changes:

```
# Load packages

require(devtools, lib.loc = "/gscratch/srlab/rpackages/")
require(BiocManager, lib.loc = "/gscratch/srlab/rpackages/")
require(methylKit, lib.loc = "/gscratch/home/yaaminiv/R/x86_64-pc-linux-gnu-library/3.6/")
require(dplyr, lib.loc = "/gscratch/srlab/rpackages/")
sessionInfo()
```

my script isn't running yet, but I will update this discussion once it starts to run.

**Reply by kubu4** [reply]
Excellent! Looks like script is still running after ~3mins... That's a good sign!
